### ISSUE TITLE
Adding solidity's default visibility to variables and functions.

### ISSUE THREAD

**Issue by Janther** [prettier-solidity/prettier-plugin-solidity#123]
One of the things that I like about [Prettier](https://github.com/prettier/prettier) is the fact that is opinionated and makes some desitions around the formatting using standard industry practices.

I want to open the discussion on whether this formatter should take care of the visibility of variables and functions when the developer forgets to add them explicitly.

The compiled bytecode of the resulting code would not be altered and the resulting code will be more descriptive.

**Comment by mattiaerre** [comment]
I'm not inclined to "add" as I'd prefer to have the code just formatted (based on the AST); moving things around is what prettier should be doing; adding language specific notations it should not be its business.

// cc @fvictorio @j-f1

**Comment by fvictorio** [comment]
Oh, this is a really hard one.

If the AST is the same, I think it's not a problem for prettier to do it. But on the other hand, this _feels_ like something that should be taken care by a linter.

**Comment by aboutlo** [comment]
IMO being explicit is always a good practice. I think having a tool that can enforce the best practice would be amazing. 
https://github.com/ConsenSys/smart-contract-best-practices/blob/master/docs/recommendations.md#explicitly-mark-visibility-in-functions-and-state-variables

**Comment by maxsam4** [comment]
FWIW In Solidity 0.5, it is mandatory to explicitly define the function visibility. Although I think that it is not something prettier should be doing by default, I already implemented this when I created the codemod for solidity 0.4 to 0.5. So, If the community decides in favor of doing this, it will be a relatively easy task. I personally will prefer to leave this out of solidity-prettier.

This is really a linter's job. There is a reason why solidity 0.5 straight up declines to compile rather than showing a warning and compiling anyway if you don't provide the visibility explicitly.
